python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -351,6 +363,20 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
     def get_autoscaling_actor_pools(self) -> List[AutoscalingActorPool]:
         return [self._actor_pool]
 
+    def update_resource_usage(self) -> None:
+        """Updates resources usage."""
+        actors = list(self._actor_pool._num_tasks_in_flight.keys())
+        for actor in actors:
+            actor_state = actor._get_local_state()
+            if actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE:
+                # If an actor is not ALIVE, it's a candidate to be marked as a
+                # restarting actor.
+                self._actor_pool.running_to_restarting(actor, actor.get_location)
+            else:
+                # If an actor is ALIVE, it's a candidate to be marked as a
+                # running actor, if not already the case.
+                self._actor_pool.restarting_to_running(actor.get_location)
```

Comment:
should use the actor handle as the key, `actor.get_location` is a method, not an object ref.

Reply:
Similar to the above comment, let's add some unit test to cover the state transitions.